File: 3754_Concatenate_Non-Zero_Digits_and_Multiply_by_Sum_I/solution.py
# class Solution:
#     def sumAndMultiply(self, n: int) -> int:
#         num = 0
#         sum_of_digits = 0

#         while n != 0:
#             digit = n % 10
#             if digit != 0:
#                 num = num * 10 + digit
#                 sum_of_digits += digit
#             n //= 10

#         return int(str(num)[::-1]) * sum_of_digits

class Solution:
    def sumAndMultiply(self, n: int) -> int:
        num, sum_of_digits, multiplier = 0, 0, 1

        while n != 0:
            digit = n % 10
            if digit != 0:
                num = digit * multiplier + num
                multiplier *= 10
                sum_of_digits += digit
            n //= 10

        return num * sum_of_digits

File: 3754_Concatenate_Non-Zero_Digits_and_Multiply_by_Sum_I/solution.c
// long long reverse(int n) {
//     long long num = 0;
//     while (n != 0) {
//         long long digit = n % 10;
//         num = num * 10 + digit;
//         n /= 10;
//     }
//     return num;
// }

// long long sumAndMultiply(int n) {
//     long long num = 0, sum = 0;
//     while (n != 0) {
//         long long digit = n % 10;
//         if (digit != 0) {
//             num = num * 10 + digit;
//             sum += digit;
//         }
//         n /= 10;
//     }
//     return reverse(num) * sum;
// }

long long sumAndMultiply(int n) {
    long long num = 0, sum = 0, multiplier = 1;
    while (n != 0) {
        long long digit = n % 10;
        if (digit != 0) {
            num = digit * multiplier + num;
            multiplier *= 10;
            sum += digit;
        }
        n /= 10;
    }
    return num * sum;
}